### PR TITLE
setup firebase-firestore

### LINKED ISSUE #3: Setup firebase and connect to REST API
- [ ] Setup firebase
- [ ] Connect instance to REST API (maybe have the API retrieve and send back some dummy data or something)

### PR DESCRIPTION
**Pull request by gokcedilek**
hey @narendrasss @jackyzha0 @jmengo @kx-chen, I added some sample firestore configuration - not sure if it's worth merging because it's just a test, but wanted to ask your opinion with a PR nevertheless. (@jackyzha0 I also had to make two small changes in the Dockerfile, 1) during the 2nd stage when running the container with a new image, instead of "scratch", I used a "alpine" base image because the former gave a "x509: certificate signed by unknown authority" error when accessing Firebase, though we can work on solving this separately to make it work with the scratch image too. 2) again during the 2nd stage, in order to be able to use the Firebase ServiceAccountKey file, I changed workdir and entrypoint commands slightly - it's probably self-explanatory enough)

### CODE REVIEW

File: backend/Dockerfile
Comment (jackyzha0) [inline backend/Dockerfile:30]: instead of using `alpine` you can add this line to add network certs and git:

```Dockerfile
RUN apk add git ca-certificates
```

File: backend/test_firestore.go
Comment (jackyzha0) [inline backend/test_firestore.go]: can we use `logrus` here for consistency? it's the log package used in `backend/main.go`

File: backend/test_firestore.go
Comment (jackyzha0) [inline backend/test_firestore.go]: maybe add a comment here explaining that this section unmarshals request JSON into the data struct

File: backend/Dockerfile
Comment (gokcedilek) [reply backend/Dockerfile:30]: it seems like the scratch image doesn't have apk installed: https://stackoverflow.com/a/40211742/11223254 
`
FROM scratch
RUN apk add git ca-certificates
`
gave: 
`
Step 11/16 : RUN apk add git ca-certificates
 ---> Running in eca7ceea97fa
OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
`

Comment (jackyzha0) [reply backend/Dockerfile:30]: ok jokes I forgot about that, there's technically a way to copy to certs over but it's very hacky. Let's leave it as `alpine` then!

**Comment by nandanmen** [comment]
Thanks for getting this started @gokcedilek and for the hasty review @jackyzha0! Appreciate the discussion so far.

**Review by jackyzha0 — APPROVED** [review]
clean stuff 🧼

**Review by nandanmen — CHANGES_REQUESTED** [review]
Code looks good @gokcedilek! Can you add a section on the README for setting up the firebase service account key and getting that working?

Once that's done, it's good to merge :)

**Comment by jackyzha0** [comment]
@narendrasss we should cut out another ticket to turn the stuff in this PR into a usable module when we get the chance to

**Comment by nandanmen** [comment]
> @narendrasss we should cut out another ticket to turn the stuff in this PR into a usable module when we get the chance to

Yup agreed! We should maybe spend a bit of time next week thinking about the possible architecture for the service.

**Review by nandanmen — APPROVED** [review]

**Comment by nandanmen** [comment]
@gokcedilek is this good to go? If it is feel free to squash and merge :)

**Comment by gokcedilek** [comment]
sounds good, I'll merge it now, once our firestore use case is more clear we can improve it! thanks for the reviews!:)

**Pull request merged**
